### PR TITLE
Add clear criteria

### PR DESCRIPTION
**Pull request by ckchiu2**
## Description

Added and exposed clearCriteria() from MaterialTable. Use the following code to clear existing criteria, including filters, sorts and search text. Page will be reset to 0.

`
tableRef.current.clearCriteria()
`

## Impacted Areas in Application

<MaterialTable />
data-manager.js

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -495,4 +495,10 @@ export type ColumnSize = {
 
 export default class MaterialTable<
   RowData extends object
-> extends React.Component<MaterialTableProps<RowData>> {}
+> extends React.Component<MaterialTableProps<RowData>> {
+  onQueryChange: (
+    query: Partial<Query<RowData>>,
+    callback?: () => void
+  ) => void;
+  clearCriteria: () => void;
```
Comment (Domino987) [inline types/index.d.ts:503]: This needs to be at the MaterialTableProps level to be consistent with everything

**Review by Domino987 — CHANGES_REQUESTED** [review]
Can we not change the prettier though this PR and only change the code that is required here?

**Comment by ckchiu2** [comment]
If you mean to revert auto formatted code like this, it is fine.
I don't know why it is auto-formatted though.

**Comment by Domino987** [comment]
> If you mean to revert auto formatted code like this, it is fine. I don't know why it is auto-formatted though.

Prettier most likely, but ok.

The prop types still need to be addressed though

**Comment by ckchiu2** [comment]
Sorry but what do you mean by prop types?
The added type is for the ref, i.e. the MaterialTable object, not for the props.

**Review by Domino987 — APPROVED** [review]

**Pull request merged**